build.gradle

```diff
@@ -690,6 +690,7 @@ project(':streams') {
     testCompile project(':core')
     testCompile project(':core').sourceSets.test.output
     testCompile libs.junit
+    testCompile 'com.github.jbellis:jamm:0.3.1'
```

Comment:
Have we decided that we will use jamm? Or is this exploratory still? If the latter, is there benefit in merging the code at this stage?